src/test/java/com/example/moviemanager/service/MovieServiceTest.java

```diff
@@ -0,0 +1,155 @@
+
+        // when
+        MovieNotFoundException thrown =
+                assertThrows(MovieNotFoundException.class, () -> movieService.findById(2));
+        String actualMessage = thrown.getMessage();
+
+        // then
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    void returns_exactly_one_movie() {
+        // given
+        Movie expectedMovie = new Movie(1, "Home Alone", "Christmas movie", 1990, 8.5);
+        when(movieRepository.findById(1)).thenReturn(Optional.of(expectedMovie));
+
+        // when
+        Movie actualMovie = movieService.findById(1);
+
+        // then
+        assertEquals(expectedMovie, actualMovie);
+    }
+
+    @Test
+    void throws_exception_when_trying_to_delete_non_existing_movie() {
+        // given
+        int movieId = 2;
+        String expectedMessage = "Could not find Movie by id - " + movieId;
+        doThrow(new MovieNotFoundException(expectedMessage)).when(movieRepository).deleteById(2);
+
+        // when
+        MovieNotFoundException thrown =
+                assertThrows(MovieNotFoundException.class, () -> movieService.deleteById(2));
+        String actualMessage = thrown.getMessage();
+
+        // then
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    void deletes_one_movie() {
+        // given
+        int movieId = 1;
+        doNothing().when(movieRepository).deleteById(movieId);
+
+        // when
+        movieService.deleteById(1);
+
+        // then
+        verify(movieRepository, times(1)).deleteById(movieId);
+        verifyNoMoreInteractions(movieRepository);
+    }
+
+    @Test
+    void saves_one_movie() {
+        // given
+        Movie expected = new Movie(1, "Home Alone", "Christmas movie", 1990, 8.5);
+        when(movieRepository.save(expected)).thenReturn(expected);
+
+        // when
+        Movie actual = movieService.save(expected);
+
+        // then
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void updates_one_movie() {
+        // given
+        int movieId = 1;
+        when(movieRepository.findById(movieId))
+                .thenReturn(Optional.of(new Movie(movieId, "Home Alone 2", "Comedy", 1992, 9.0)));
+
+        Movie expected = new Movie(5, "Home Alone", "Christmas movie", 1990, 8.5);
+        expected.setId(movieId);
+        when(movieRepository.save(expected)).thenReturn(expected);
+
+        // when
+        Movie actual = movieService.update(movieId, expected);
+
+        // then
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void throws_exception_when_trying_to_update_non_existing_movie() {
+        // given
+        int movieId = 5;
+        Movie movie = new Movie(1, "Home Alone", "Christmas movie", 1990, 8.5);
+        String expectedMessage = "Update failed. Could not find movie by id - " + movieId;
+        doThrow(new MovieNotFoundException(expectedMessage)).when(movieRepository).findById(5);
```

Comment:
How is it possible that movie repository could throw such exception? It is your service exception and repository does not know anything about it. This test looks wrong

Reply:
Fixed all tests in this class with incorrect return of movie repository.